lib/ansible/modules/storage/dellemc/dellemc_pmax_createsg.py

```diff
@@ -0,0 +1,200 @@
+        num_vols: 1
+        password: "{{password}}"
+        sgname: "{{sgname}}"
+        slo: Diamond
+        srp_id: SRP_1
+        unispherehost: "{{unispherehost}}"
+        universion: "{{universion}}"
+        user: "{{user}}"
+        verifycert: "{{verifycert}}"
+        vol_size: 1
+        workload: None
+        volumeIdentifier: 'REDO'
+'''
+RETURN = '''
+'''
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    changed = False
+    module = AnsibleModule(
+        argument_spec=dict(
+            sgname=dict(type='str', required=True),
+            unispherehost=dict(required=True),
+            universion=dict(type='int', required=False),
+            verifycert=dict(type='bool', required=True),
+            user=dict(type='str', required=True),
+            password=dict(type='str', required=True, no_log=True),
+            array_id=dict(type='str', required=True),
+            srp_id=dict(type='str', required=False),
+            slo=dict(type='str', required=False),
+            workload=dict(type='str', required=False),
+            num_vols=dict(type='int', required=True),
+            vol_size=dict(type='int', required=True),
+            cap_unit=dict(type='str', required=True, choices=['GB',
+                                                              'TB',
+                                                              'MB', 'CYL']),
+            volumeIdentifier=dict(type='str', required=False)))
+    try:
+        import PyU4V
+    except:
+        module.fail_json(
+            msg='Requirements not met PyU4V is not installed, please install '
+                'via PIP')
+        module.exit_json(changed=changed)
+
+    conn = PyU4V.U4VConn(server_ip=module.params['unispherehost'], port=8443,
+                         array_id=module.params['array_id'],
+                         verify=module.params['verifycert'],
+                         username=module.params['user'],
+                         password=module.params['password'],
+                         u4v_version=module.params['universion'])
+    dellemc = conn.provisioning
+    # Compile a list of existing storage groups.
+    sglist = dellemc.get_storage_group_list()
+    # Check if Storage Group already exists
+    if module.params['sgname'] not in sglist:
+        dellemc.create_storage_group(srp_id='SRP_1',
+                                     sg_id=module.params['sgname'],
+                                     slo=module.params['slo'],
+                                     num_vols=module.params['num_vols'],
+                                     vol_size=module.params['vol_size'],
+                                     cap_unit=module.params['cap_unit'],
+                                     workload=None,
+                                     vol_name=module.params['volumeIdentifier']
+                                     )
+        changed = True
+    else:
+        module.fail_json(msg='Storage Group Already Exists')
```

Comment:
@AnilDegwekar so if I change the else statement here to exit the module with changed=false and still success this should be ok?

Reply:
@AnilDegwekar looking at this again, my intent here is to fail if the storage group exists.  I really dont' want to be able to continue if the storage group name already exists as this could lead to a masking view being created later presenting storage volumes to multiple hosts causing data loss.

Reply:
From Ansible documentation:

Modules should be idempotent, that is, running a module multiple times in a sequence should have the same effect as running it just once. One way to achieve idempotency is to have a module check whether its desired final state has already been achieved, and if that state has been achieved, to exit without performing any actions. If all the modules a playbook uses are idempotent, then the playbook itself is likely to be idempotent, so re-running the playbook should be safe.

Reply:
https://docs.ansible.com/ansible/latest/user_guide/playbooks_intro.html